src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -601,22 +609,23 @@ private static String getStringInsideTag(String result, DetailNode detailNode) {
      *
      * @param ast The Javadoc root node.
      * @param period The configured period symbol.
-     * @return The first sentence up to and excluding the period, or null if no ending was found.
+     * @return The first sentence up to and including the period,
+     * or empty string if no ending was found.
```

Comment:
This is not good, as you force all users to read javadoc to understand how it works.
Please return Optional it will be very explicit on how method works.

Reply:
done

Reply:
do we need to do the same with `findSentenceEnding` as well ?

Reply:
yes, please.